### ISSUE TITLE
DOC Add more examples

### ISSUE THREAD

**Issue by rth** [pyodide/pyodide#1498]
It might be helpful to have more examples in the doc (in general, not necessarily for this release).

For instance, what we could do, is,
 - have an `examples` folder with some examples as HTML files
 - in the documentation, have `usage/examples/` with an auto-generated list of pages (each with one those examples). sphinx-gallery could help there, but I'm not sure it would really make sense for .html files.
 - bonus points if we could run the example on the same page (clearly not a priority now)

Though the issue is that then we would also have tests for these examples to make sure they don't get broken.

**Comment by hoodmane** [comment]
Once we get the basics for this set up, we should probably solicit example ideas / contributions from people like @dmondev.

**Comment by dmondev** [comment]
> Once we get the basics for this set up, we should probably solicit example ideas / contributions from people like @dmondev.

Sure, @hoodmane , glad to help !!

I went ahead and created this: https://codepen.io/pyodide/pens/public

Those are two simple/crappy examples, but you get the idea.

It's another way of putting examples out there. Doesn't have the advantages of source/control/CI, but it's a nice way to have a sample gallery out there. Not sure if they have free pro/team licenses for open source projects, I've sent an email to support to check.

Again, forgive me for using the pyodide name, I just really like things in proper namespaces :)

Yet another option is https://pages.github.com/. You're already in GitHub, so it might be easier to go from there.

**Comment by hoodmane** [comment]
Thanks for this @dmondev.

> forgive me for using the pyodide name

It's better that you took it than someone random. Ideally you should share the credentials with pyodide/core.

@rth We should probably reserve the name Pyodide on pypi.
https://pypi.org/search/?q=pyodide

Pyodide is taken on npm. We should write to @ajkachnic and ask for him to give it to us.
https://www.npmjs.com/package/pyodide

**Comment by hoodmane** [comment]
@dmondev @rth Any other places you think we might want to reserve the name Pyodide?

**Comment by dmondev** [comment]
Just let me know where to send the credentials to.

I also reserved the organization name at npm so if you want to have multiple packages at @pyodide, as well.

I can't think of other places to reserve the name now.

**Comment by ajkachnic** [comment]
Hello, I was mentioned earlier in the thread. I'd be happy to transfer the name `pyodide` on npm, since I haven't touched the fork in quite a while. Is there a way for me to transfer ownership? How would I go about this?

**Comment by hoodmane** [comment]
Could you transfer it to my account (hoodmane) for now? The instructions are here:

https://docs.npmjs.com/transferring-a-package-from-a-user-account-to-another-user-account

**Comment by hoodmane** [comment]
By the way, the picture in that help article isn't quite right: there is no "admin" tab, you want the "settings" tab on the far right.

**Comment by ajkachnic** [comment]
Okay, I've invited you as a maintainer and removed myself. Is there anything else I have to do?

**Comment by hoodmane** [comment]
That's it, thanks!

**Comment by jokoon** [comment]
Honestly I would gladly, really like to see a simple example of DOM events stuff done in pyodide.

I liked brython but it's a little slow. I'm wondering if I could see an example of "replacing" js with python to script a webpage.

The only real example I'm looking for would be one that includes this:

```
    from js import document
    x = document.getElementById("myElement")
```

with possibly an event callback.

**Comment by carlkl** [comment]
Hi there,

There is a nice example I found on the net that could be useful as an example of using pyodide for simulation tasks:

https://www.itp.uni-hannover.de/de/ag/weimer/lehre/ideal-gas-simulator/rapid-python-gui-development-with-html5-and-pyodide/
https://www.itp.uni-hannover.de/fileadmin/itp/ag/weimer/idealgas/idealgas.tar.gz and https://www.itp.uni-hannover.de/de/ag/weimer/lehre/ (three simulation demos)

contact person https://www.itp.uni-hannover.de/de/institut/personen/pd-dr-hendrik-weimer/ 

Having a pyodide showcase with demos like this would be very welcome from the user's point of view.

**Comment by rth** [comment]
Going back to this topic, I would like to put more effort into examples for the next release. 

In terms of tooling, I think we need,
 1. interactive examples that don't require running a server. The selling point of Pyodide is that it can be tested in the browser without installing anything. If one has to have Python to run a webserver that's a less compelling demo.
 2. versioned in git. Otherwise, the maintenance (e.g. updating  Pyodide versions) is just too time-consuming.
 3. with CI, otherwise examples will get broken over time.
 4. ideally can be embedded into our documentation, so that we could have a gallery of examples similar to e.g. https://sphinx-gallery.github.io/stable/auto_examples/index.html

I have been playing with JSfiddle with GH integration in https://github.com/pyodide/pyodide-blog/pull/13#issuecomment-1015889592, it has a number of restrictions and the documentation is not great,  however, I think it would verify all of the above requirements. Regarding the codepen examples illustrated earlier in this discussion, they look very nice but the lack of git versioning is a deal-breaker IMO for the long-term maintenance of those. Of course, if anyone knows of a solution that would check the above boxes I would be happy to consider it. So far for the solutions, I looked into, aside from JSfiddle, do not seem to support loading files from Github.

If we go with JsFiddle, we could,
1. Put examples in subfolders of https://github.com/pyodide/pyodide-examples . We cant use the main repo because it requires the default branch to be named `master` (a known bug). In pyodide-examples we would have a hierarchy of folders, with inside, the files `demo.html`, `demo.js`, `demo.description` (see the first link in my comment for examples). That's what JsFiddle needs to work.
2. The downside is those cannot be launched with a local server. However, we can have some automated tool to assemble them into an `index.html` with valid and complete HTML.
3. We could also setup CI to load that HTML file in selenium and check that there are no errors in the console.
4. And we could have a sphinx plugin that auto-generates the list of examples from that repo, and create a new page with embedded jsfiddle iframe for each.

The only thing I'm not sure about is versioning. JsFiddle <-> GH integration only works on the main branch, which means we can't use release branches there for past releases. At the same time if we integrated it into the docs, those examples should not break in the future. So that would mean that pyodide-examples  would have a subfolder per version, which is not great with respect to navigation on GH. Not sure what would be best there.

WDYT?

**Comment by ryanking13** [comment]
What I've imagined are some interactive tutorial pages (+ some example galleries) similar to:

- https://svelte.dev/tutorial/basics
- https://go.dev/tour/welcome/1

which might be hosted with, for example, Github Pages, without running a server.
Though it would be more painful in the beginning compared to embedding JSFiddle or CodePen...

**Comment by rth** [comment]
I agree that having a custom solution that could allow storing examples on github (ideally in the main repo) and deploying them ourselves as part of the documentation or as a separate examples website would be even better. Tough the question is can we find an open-source implementation that we can use/adapt for it. If we have to re-implement it from scratch that's a pretty large side project.

**Comment by rth** [comment]
For interactive examples embedded in the docs https://elilambnz.github.io/react-py/docs/examples/basic-example also looks nice. But I don't know what it would take to integrate that react widget into sphinx documentation.

**Comment by sathvikshetty520** [comment]
Hi, I would like to contribute an example demonstrating how Pyodide interacts with the DOM (using `from js import document`) and handles events in the browser. Would this be appropriate for this issue?

**Comment by ryanking13** [comment]
@sathvikshetty520 Sure, PR welcomed
